cli_tools/google-osconfig-agent/patch/patch_run.go

```diff
@@ -16,38 +16,68 @@ package patch
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"time"
 
-	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
-	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
+	"cloud.google.com/go/compute/metadata"
+	osconfig "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
+	api "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/config"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/logger"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/tasker"
 	"github.com/golang/protobuf/jsonpb"
 	"google.golang.org/api/option"
 )
 
+const (
+	identityTokenPath = "instance/service-accounts/default/identity?audience=osconfig.googleapis.com&format=full"
+)
+
+type patchStep int
+
+// These patch steps are ordered to allow us to skip steps that have already been completed. Progress should
+// always move forward, even if we need to retry a step.
+const (
+	unknown patchStep = iota
```

Comment:
Reduce these to just the ones we act on currently, then add TODO to refactor into steps and step types